apps/api/src/services/competition-manager.service.ts

```diff
@@ -532,7 +534,34 @@ export class CompetitionManager {
    */
   async getLeaderboard(competitionId: string) {
     try {
-      // First try to get from the competitions_leaderboard table
+      const competition = await findById(competitionId);
+      const competitionStatus = competition?.status;
+
+      // Default case: `COMPETITION_STATUS.PENDING` will have no leaderboard—so we just return the agents
```

Comment:
this logic was added because the old logic would try to make some queries and perform portfolio calculations when a competition is "pending". 

a pending comp doesnt have trading data. we just return the agents in order of their global rank, and unranked agents come afterward.